src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/MessagePackHubProtocolTest.java

```diff
@@ -0,0 +1,914 @@
+package com.microsoft.signalr;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.lang.reflect.Type;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import okio.ByteString;
+
+import org.junit.jupiter.api.Test;
+
+class MessagePackHubProtocolTest {
+    private MessagePackHubProtocol messagePackHubProtocol = new MessagePackHubProtocol();
+
+    @Test
+    public void checkProtocolName() {
+        assertEquals("messagepack", messagePackHubProtocol.getName());
+    }
+
+    @Test
+    public void checkVersionNumber() {
+        assertEquals(1, messagePackHubProtocol.getVersion());
+    }
+
+    @Test
+    public void checkTransferFormat() {
+        assertEquals(TransferFormat.BINARY, messagePackHubProtocol.getTransferFormat());
+    }
+    
+    @Test
+    public void verifyWriteInvocationMessage() {
+        InvocationMessage invocationMessage = new InvocationMessage(null, null, "test", new Object[] { 42 }, null);
+        ByteBuffer result = messagePackHubProtocol.writeMessage(invocationMessage);
+        byte[] expectedBytes = {0x0C, (byte) 0x96, 0x01, (byte) 0x80, (byte) 0xC0, (byte) 0xA4, 0x74, 0x65, 0x73, 
```

Comment:
Why are some values cast to `(byte)` and others aren't?

Reply:
Java considers bytes 0x80 & above (in this format) to be ints. I could convert all of these to decimal, but I thought it was more readable keeping them in this format.